### PR TITLE
Add options to multi client

### PR DESCRIPTION
**Pull request by rentziass**
This was added while working on TLS, and this being a necessity (for proxy work too) I wanted to avoid pushing another giant PR 😄

### CODE REVIEW

**Review by nikola-jokic — APPROVED** [review]
LGTM

File: github/actions/multi_client.go [hunk 1]
```diff
@@ -66,8 +66,10 @@ func (m *multiClient) GetClientFor(ctx context.Context, githubConfigURL string,
 	client, err := NewClient(
 		githubConfigURL,
 		&creds,
-		WithUserAgent(m.userAgent),
-		WithLogger(m.logger),
+		append([]ClientOption{
+			WithUserAgent(m.userAgent),
+			WithLogger(m.logger),
+		}, options...)...,
```
Comment (TingluoHuang) [inline github/actions/multi_client.go:72]: do we need these many `...`? 😄

Comment (nikola-jokic) [reply github/actions/multi_client.go:72]: You have to break them to variadic arguments by doing that
Only thing that can be done is to split it as:
```go
options := []ClientOption{
			WithUserAgent(m.userAgent),
			WithLogger(m.logger),
		}, opts...)
```
And then call it with `options...` 
I'm okay with both :smile:

Comment (rentziass) [reply github/actions/multi_client.go:72]: The idea was that that block wouldn't be changed very easily 😄 but yeah, what Nikola said :)

**Pull request merged**